src/includes/class_reservationPermissions.php

```diff
@@ -1,76 +1,62 @@
 <?php
 class reservationPermissions {
 
-  public function getRecords($id = null){
-      try {
-          // call engine
-          $engine    = EngineAPI::singleton();
-          $localvars = localvars::getInstance();
-          $db        = db::get($localvars->get('dbConnectionName'));
-          $sql       = "SELECT * FROM `reservePermissions`";
-          $validate  = new validate;
+  private $localvars;
+  private $engine;
+  private $this->db;
+  private $validate;
+
+  function __construct() {
+    $this->engine    = EngineAPI::singleton();
+    $this->localvars = localvars::getInstance();
+    $this->db        = db::get($this->localvars->get('dbConnectionName'));
+    $this->validate  = new validate;
+  }
 
-          if (isset($id)) {
-           $id        = dbSanitize($id);
-          }
+  public function getRecords($id = null){
+    $sql = "SELECT * FROM `reservePermissions`";
+    try {
+      if(!isnull($id) && !$this->validate->integer($id)){
+        throw new Exception("Invalid ID provided.");
+      }
 
-          // test to see if Id is present and valid
-          if(!isnull($id) && $validate->integer($id)){
-              $sql .= sprintf('WHERE id = %s LIMIT 1', $id);
-          }
+      if(!isnull($id)){
+        $sql .= "WHERE id = ? LIMIT 1";
+      }
 
```

Comment:
we NEVER, EVER, NEVER, put a variable into a sql statement like this. never. This leads to sql injection attacks. 

If we HAVE to put a variable into a string it MUST be escaped like: $this->db->escape($id) ... this takes care of the db sanitization. Never sanitie it separately. always do it in the sprintf statement. That way there is absolutely no confusion as to whether it has been sanitized. Otherwise someone looking at the code my not know if it has been, or, if you modify it after checking it, those modifications aren't being checked.